src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -72,15 +71,16 @@ public class HeaderDecoder : IRlpValueDecoder<BlockHeader>, IRlpStreamDecoder<Bl
                 blockHeader.BaseFeePerGas = decoderContext.DecodeUInt256();
             }
 
-            int itemsRemaining = decoderContext.PeekNumberOfItemsRemaining(null, 2);
+            int itemsRemaining = decoderContext.PeekNumberOfItemsRemaining(null, 3);
             if (itemsRemaining > 0 &&
                 decoderContext.PeekPrefixAndContentLength().ContentLength == Keccak.Size)
             {
                 blockHeader.WithdrawalsRoot = decoderContext.DecodeKeccak();
 
-                if (itemsRemaining == 2 && decoderContext.Position != headerCheck)
+                if (itemsRemaining == 3 && decoderContext.Position != headerCheck)
                 {
-                    blockHeader.ExcessDataGas = decoderContext.DecodeUInt256();
+                    blockHeader.DataGasUsed = decoderContext.DecodeULong();
+                    blockHeader.ExcessDataGas = decoderContext.DecodeULong();
```

Comment:
There is inconsistency - in rlpstream context there is `allowLeadingZeroBytes: false` and here it is missing